### ISSUE TITLE
1.10 Issue Burndown

### ISSUE THREAD

**Issue by jberkus** [kubernetes/sig-release#86]
This issue is for status updates on 1.10 issues for release tracking.

**Comment by jberkus** [comment]
Status as of 2/21, 1 day into Code Slush.

Detail behind this report can be found [on the tracking spreadsheet](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit?usp=sharing)

1 day after Code Slush, here is issue status.

## Bump

Recommend bumping these issues from the milestone based on inactivity and lack of responsiveness by SIGs/issue owner: 11 issues.

* [CronJob controller should use shared informers](https://github.com/kubernetes/kubernetes/issues/17130)
* [move kube-proxy into a daemonset](https://github.com/kubernetes/kubernetes/issues/23225)
* [Document what docker image (Dockerfile) features we support](https://github.com/kubernetes/kubernetes/issues/30039)
* [e2e tests for cloud-controller-manager](https://github.com/kubernetes/kubernetes/issues/44975)
* [Review Cluster Autoscaler code for Azure](https://github.com/kubernetes/kubernetes/issues/47511)
* [Graduate the kubeletconfig API to beta](https://github.com/kubernetes/kubernetes/issues/53084)
* [RFC: Validate Docker against the Docker API versions](https://github.com/kubernetes/kubernetes/issues/53221)
* [Remove docker dependency on Kubelet startup](https://github.com/kubernetes/kubernetes/issues/54090)
* [Support out-of-tree authentication providers](https://github.com/kubernetes/kubernetes/issues/55968)
* [Communicate etcd2 support deprecation timeline](https://github.com/kubernetes/kubernetes/issues/57354)
* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [Don't duplicate status in audit events](https://github.com/kubernetes/kubernetes/issues/60108)
* [audit.Event.RequestObject underspecified for patch requests](https://github.com/kubernetes/kubernetes/issues/60110)
* [Remove the PersistentVolumeLabel Admission Controller](https://github.com/kubernetes/kubernetes/issues/52617)

## Wait

Status of issues/PRs has been queried, currently waiting for response from owners: 9 issues

* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)
* [Support a Vault based KMS provider for envelope encryption of resources in a cluster](https://github.com/kubernetes/kubernetes/issues/49817)
* [ConfigMaps and Secrets mounted with subPath do not update when changed](https://github.com/kubernetes/kubernetes/issues/50345)
* [kubernetes GPU device plugin for Ubuntu and Debian images](https://github.com/kubernetes/kubernetes/issues/54011)
* [Delete in-tree support for Nvidia GPUs](https://github.com/kubernetes/kubernetes/issues/54012)
* [Job backoff limit workings when parallelism > 1](https://github.com/kubernetes/kubernetes/issues/56853)
* [RFC: Some thoughts about node e2e tests](https://github.com/kubernetes/kubernetes/issues/59001)
* [Document upgrade and downgrade steps for etcd 3.2 upgrade](https://github.com/kubernetes/kubernetes/issues/59221)

## Keep

1.10 issues which appear to be in progress and headed towards resolution by Code Freeze: 19 issues

* [CRI: Support CRI log stats](https://github.com/kubernetes/kubernetes/issues/55905)
* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [Use of resourceVersion=0 in reflectors for initial sync breaks pod safety when more than one api server](https://github.com/kubernetes/kubernetes/issues/59848)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [Add support for accessing Ceph RBD with the rbd-nbd client](https://github.com/kubernetes/kubernetes/issues/32266)
* [[audit] Restore audit logging in the scalability tests](https://github.com/kubernetes/kubernetes/issues/53020)
* [After 1.8, scheduler could reject unknown extended resource names](https://github.com/kubernetes/kubernetes/issues/53616)
* [Kubelet: Resolve paths from dynamic config payloads against unpacked config file location](https://github.com/kubernetes/kubernetes/issues/55645)
* [add PV resize support for azure disk](https://github.com/kubernetes/kubernetes/issues/56463)
* [Kubernetes Container Runtime Interface (CRI) doesn't support WindowsContainerConfig and WindowsContainerResources](https://github.com/kubernetes/kubernetes/issues/56734)
* [Graduate the Lease Endpoint Reconciler to beta in v1.10](https://github.com/kubernetes/kubernetes/issues/57617)
* [Add a subresource for Node.ConfigSource?](https://github.com/kubernetes/kubernetes/issues/58362)
* [CRI: Implement container log rotation](https://github.com/kubernetes/kubernetes/issues/58823)
* [Invalid capacity 0 on windows image filesystem](https://github.com/kubernetes/kubernetes/issues/59742)
* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [Node's providerID is wrong for azure vmss when using instance metadata](https://github.com/kubernetes/kubernetes/issues/59855)
* [[e2e test failure] each node by dropping all outbound packets for a while and ensure they function afterwards](https://github.com/kubernetes/kubernetes/issues/59889)
* [[test failure] verify.openapi-spec](https://github.com/kubernetes/kubernetes/issues/59895)
* [Cut and Vendor cAdvisor v0.29.0](https://github.com/kubernetes/kubernetes/issues/60052)

## Tracking

These are tracking issues.  In a few cases, they link to multiple issues which are NOT currently marked with the 1.10 milestone.  6 issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [CRI: support non-builtin container runtimes](https://github.com/kubernetes/kubernetes/issues/47872)
* [Move Priority and Preemption to Beta](https://github.com/kubernetes/kubernetes/issues/57471)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)
* [Standalone Azure cloud provider](https://github.com/kubernetes/kubernetes/issues/50752)
* [Move `KubeControllerManagerConfiguration` to `pkg/controller/apis/`](https://github.com/kubernetes/kubernetes/issues/57618)

**Comment by liggitt** [comment]
`Graduate the kubeletconfig API to beta` was completed in 1.10

**Comment by liggitt** [comment]
`Support out-of-tree authentication providers` design is merged, implementing PR is under review and planned to merge this week

**Comment by jdumars** [comment]
All SIG-Azure issues have been curated.

**Comment by jberkus** [comment]
Status as of 2/23.  All issue owners have been reminded of the need to add status/approved-for-milestone.  11 issues were removed from the milestone or resolved.  3 new issues were added.  Total of 40 issues.

# Bump

Recommend bumping these issues from the milestone based on inactivity and lack of responsiveness by SIGs/issue owner: 8 issues

* [Document what docker image (Dockerfile) features we support](https://github.com/kubernetes/kubernetes/issues/30039)
* [e2e tests for cloud-controller-manager](https://github.com/kubernetes/kubernetes/issues/44975)
* [Review Cluster Autoscaler code for Azure](https://github.com/kubernetes/kubernetes/issues/47511)
* [Remove the PersistentVolumeLabel Admission Controller](https://github.com/kubernetes/kubernetes/issues/52617)
* [RFC: Validate Docker against the Docker API versions](https://github.com/kubernetes/kubernetes/issues/53221)
* [kubernetes GPU device plugin for Ubuntu and Debian images](https://github.com/kubernetes/kubernetes/issues/54011)
* [Remove docker dependency on Kubelet startup](https://github.com/kubernetes/kubernetes/issues/54090)
* [add PV resize support for azure disk](https://github.com/kubernetes/kubernetes/issues/56463)
* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)

# Wait

Issue Owner/SIG has been queried, waiting for response.  8 issues

* [Add support for accessing Ceph RBD with the rbd-nbd client](https://github.com/kubernetes/kubernetes/issues/32266)
* [ConfigMaps and Secrets mounted with subPath do not update when changed](https://github.com/kubernetes/kubernetes/issues/50345)
* [After 1.8, scheduler could reject unknown extended resource names](https://github.com/kubernetes/kubernetes/issues/53616)
* [Delete in-tree support for Nvidia GPUs](https://github.com/kubernetes/kubernetes/issues/54012)
* [Graduate the Lease Endpoint Reconciler to beta in v1.10](https://github.com/kubernetes/kubernetes/issues/57617)
* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)
* [RFC: Some thoughts about node e2e tests](https://github.com/kubernetes/kubernetes/issues/59001)
* [Don't duplicate status in audit events](https://github.com/kubernetes/kubernetes/issues/60108)
* [audit.Event.RequestObject underspecified for patch requests](https://github.com/kubernetes/kubernetes/issues/60110)
* [Device Plugin failure handling in kubelet is racy](https://github.com/kubernetes/kubernetes/issues/60176)

# Keep

Issue is approved by SIG for 1.10 or looks likely to be approved/completed, or looks like a blocking bug against 1.10: 14 issues.

* [[audit] Restore audit logging in the scalability tests](https://github.com/kubernetes/kubernetes/issues/53020)
* [Kubelet: Resolve paths from dynamic config payloads against unpacked config file location](https://github.com/kubernetes/kubernetes/issues/55645)
* [Support out-of-tree authentication providers](https://github.com/kubernetes/kubernetes/issues/55968)
* [Kubernetes Container Runtime Interface (CRI) doesn't support WindowsContainerConfig and WindowsContainerResources](https://github.com/kubernetes/kubernetes/issues/56734)
* [Job backoff limit workings when parallelism > 1](https://github.com/kubernetes/kubernetes/issues/56853)
* [Add a subresource for Node.ConfigSource?](https://github.com/kubernetes/kubernetes/issues/58362)
* [Document upgrade and downgrade steps for etcd 3.2 upgrade](https://github.com/kubernetes/kubernetes/issues/59221)
* [Invalid capacity 0 on windows image filesystem](https://github.com/kubernetes/kubernetes/issues/59742)
* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [Use of resourceVersion=0 in reflectors for initial sync breaks pod safety when more than one api server](https://github.com/kubernetes/kubernetes/issues/59848)
* [[test failure] verify.openapi-spec](https://github.com/kubernetes/kubernetes/issues/59895)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [ Windows containers creation failed because of rslave mounts](https://github.com/kubernetes/kubernetes/issues/60274)
* [Move Taints and Tolerations to GA](https://github.com/kubernetes/kubernetes/issues/60281)

# Tracking

Tracking issues which have features/bugs in 1.10:

* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [CRI: support non-builtin container runtimes](https://github.com/kubernetes/kubernetes/issues/47872)
* [Move Priority and Preemption to Beta](https://github.com/kubernetes/kubernetes/issues/57471)
* [Move `KubeControllerManagerConfiguration` to `pkg/controller/apis/`](https://github.com/kubernetes/kubernetes/issues/57618)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)

**Comment by jberkus** [comment]
# Burndown report as of scheduled Code Freeze 2/26

Complete issue tracker is [a spreadsheet](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=552625354).

## Red

These issues represent what look like blockers for 1.10, and do not have a PR.  One has had no attention at all.

* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [Use of resourceVersion=0 in reflectors for initial sync breaks pod safety when more than one api server](https://github.com/kubernetes/kubernetes/issues/59848)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [[failing test] daemonset-upgrade in 1.9-master upgrade jobs](https://github.com/kubernetes/kubernetes/issues/60437)

## Yellow

These issues have one or more PRs.  However, either the PRs aren't in good shape (such as failing tests or review) or don't completely resolve the issue.

* [After 1.8, scheduler could reject unknown extended resource names](https://github.com/kubernetes/kubernetes/issues/53616)
* [Remove docker dependency on Kubelet startup](https://github.com/kubernetes/kubernetes/issues/54090)
* [Support out-of-tree authentication providers](https://github.com/kubernetes/kubernetes/issues/55968)
* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [Device Plugin failure handling in kubelet is racy](https://github.com/kubernetes/kubernetes/issues/60176)
* [OSS gke tests fails check APIReachability](https://github.com/kubernetes/kubernetes/issues/60380)
* [https://github.com/kubernetes/kubernetes/issues/60381](https://github.com/kubernetes/kubernetes/issues/60381)

## Green

These issues have a PR in good shape and are expected to merge and close within a couple of days.

* [[audit] Restore audit logging in the scalability tests](https://github.com/kubernetes/kubernetes/issues/53020)
* [Kubernetes Container Runtime Interface (CRI) doesn't support WindowsContainerConfig and WindowsContainerResources](https://github.com/kubernetes/kubernetes/issues/56734)
* [Document upgrade and downgrade steps for etcd 3.2 upgrade](https://github.com/kubernetes/kubernetes/issues/59221)
* [[test failure] verify.openapi-spec](https://github.com/kubernetes/kubernetes/issues/59895)
* [DaemonSet should ignore the `unschedulable` field of a node](https://github.com/kubernetes/kubernetes/issues/60163)
* [Move Taints and Tolerations to GA](https://github.com/kubernetes/kubernetes/issues/60281)

## Tracking

These are tracking issues for features being dealt with in multiple steps/releases/PRs.

* [Move `KubeControllerManagerConfiguration` to `pkg/controller/apis/`](https://github.com/kubernetes/kubernetes/issues/57618)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)

## Bump

These issues are expected to get bumped out of 1.10 once automation kicks in tommorrow, as they are not approved or are low-priority.

* [Document what docker image (Dockerfile) features we support](https://github.com/kubernetes/kubernetes/issues/30039)
* [e2e tests for cloud-controller-manager](https://github.com/kubernetes/kubernetes/issues/44975)
* [Remove the PersistentVolumeLabel Admission Controller](https://github.com/kubernetes/kubernetes/issues/52617)
* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)
* [Move `KubeControllerManagerConfiguration` to `pkg/controller/apis/`](https://github.com/kubernetes/kubernetes/issues/57618)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)
* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)
* [RFC: Some thoughts about node e2e tests](https://github.com/kubernetes/kubernetes/issues/59001)
* [Image Manager should return a copy of image list to avoid data race.](https://github.com/kubernetes/kubernetes/issues/60443)
* [Add support for accessing Ceph RBD with the rbd-nbd client](https://github.com/kubernetes/kubernetes/issues/32266)

**Comment by jdumars** [comment]
Thank you so much! This takes the organizational cake.

**Comment by jberkus** [comment]
As of today, we have 27 issues open against 1.10, although some of them (4) would have been removed by automation if we hadn't run out of github tokens.  I don't have a good comparison against prior releases because I've discovered math errors in the devstats charts.  Working on that.  Closest comparison is 20 open issues 4 days after Code Freeze for 1.9.

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) and is up to date as of this afternoon.  Note the three "NA" issues; these are recent bugs and/or test failures which *look* like 1.10 issues, but have not been confirmed by their respective SIGs.

## Red

Issues with no PR, or no complete PR, which cannot be easily kicked out of 1.10.

* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [Kubernetes is vulnerable to stale reads, violating critical pod safety guarantees](https://github.com/kubernetes/kubernetes/issues/59848)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

* [RFC: Some thoughts about node e2e tests](https://github.com/kubernetes/kubernetes/issues/59001)
* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [Improve k8s support for multizone PDs](https://github.com/kubernetes/kubernetes/issues/59988)
* [DaemonSet should ignore the `unschedulable` field of a node](https://github.com/kubernetes/kubernetes/issues/60163)
* [https://github.com/kubernetes/kubernetes/issues/60381](https://github.com/kubernetes/kubernetes/issues/60381)
* [[failing test] daemonset-upgrade in 1.9-master upgrade jobs](https://github.com/kubernetes/kubernetes/issues/60437)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [[failing test] Nginx should conform to Ingress spec](https://github.com/kubernetes/kubernetes/issues/60588)
* [Failing to list pods with selector `spec.nodeName!= ""`](https://github.com/kubernetes/kubernetes/issues/60507)

## Green

Issues with an approved PR which is just waiting for labels, release notes, or automation.

* [Document upgrade and downgrade steps for etcd 3.2 upgrade](https://github.com/kubernetes/kubernetes/issues/59221)
* [[audit] Restore audit logging in the scalability tests](https://github.com/kubernetes/kubernetes/issues/53020)
* [Support out-of-tree authentication providers](https://github.com/kubernetes/kubernetes/issues/55968)
* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)
* [[test failure] verify.openapi-spec](https://github.com/kubernetes/kubernetes/issues/59895)
* [kubectl completion failed to list file names](https://github.com/kubernetes/kubernetes/issues/60517)
* [Server side print returns generic information for deployments](https://github.com/kubernetes/kubernetes/issues/60488)

## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)

## Kick

Issues which are waiting for automation to kick them out of the milestone.

* [e2e tests for cloud-controller-manager](https://github.com/kubernetes/kubernetes/issues/44975)
* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)

**Comment by jberkus** [comment]
# Burndown report as of 3/1 around 1pm Pacific 

Complete issue tracker is [a spreadsheet](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=552625354).

We currently have around 21 "real" issues, excluding a few which will be dropped by automation as soon as the patched munger catches up.

## Red

These issues represent what look like blockers for 1.10, and do not have a PR.

* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [Kubernetes is vulnerable to stale reads, violating critical pod safety guarantees](https://github.com/kubernetes/kubernetes/issues/59848)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [DaemonSet should ignore the `unschedulable` field of a node](https://github.com/kubernetes/kubernetes/issues/60163)
* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
* [Failing to list pods with selector `spec.nodeName!= ""`](https://github.com/kubernetes/kubernetes/issues/60507)

## Yellow

These issues have one or more PRs.  However, either the PRs need review (such as failing tests or review) or don't completely resolve the issue.

* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [Improve k8s support for multizone PDs](https://github.com/kubernetes/kubernetes/issues/59988)
* [https://github.com/kubernetes/kubernetes/issues/60381](https://github.com/kubernetes/kubernetes/issues/60381)
* [[failing test] daemonset-upgrade in 1.9-master upgrade jobs](https://github.com/kubernetes/kubernetes/issues/60437)
* [kubectl completion failed to list file names](https://github.com/kubernetes/kubernetes/issues/60517)
* [[failing test] Nginx should conform to Ingress spec](https://github.com/kubernetes/kubernetes/issues/60588)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [[job flake] kubelet-master](https://github.com/kubernetes/kubernetes/issues/60604)
* [Mount propagation moved to beta, comment not updated](https://github.com/kubernetes/kubernetes/issues/60657)

## Green

These issues have a PR in good shape and are expected to merge and close within a couple of days.

* [Support out-of-tree authentication providers](https://github.com/kubernetes/kubernetes/issues/55968)
* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)

## Tracking

These are tracking issues for features being dealt with in multiple steps/releases/PRs.

* [v1.10 known issues / FAQ accumulator](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)

**Comment by jberkus** [comment]
As of today, we have 21 issues open against the 1.10 milestone, which is the same as yesterday.  However, several of those issues have moved
from yellow to green status because of PRs being approved/fixed, so we can expect a drop in the number of issues over the weekend.

On the down side, several issues are of special concern, as they represent severe problems which may throw off the release schedule.  These are all
in the Red section and detailed there.

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) 
and is up to date as of this afternoon.  

## Red

Issues with no PR, or no complete PR, which cannot be easily taken out of 1.10 or represent major regressions.

* [Kubernetes is vulnerable to stale reads, violating critical pod safety guarantees](https://github.com/kubernetes/kubernetes/issues/59848)

The Stale Reads issue is a potentially major issue, affecting all supported versions of Kubernetes, without a clear solution that
doesn't produce a major performance regression.  Depending on how fixing it goes, we may have to punt on it for 1.10.0 and wait for
a fix in 1.10.1.

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)

These two are really the same issue, and show what may be a very large performance regression in 1.10 even without a Stale Reads fix.
Waiting on test framework to find out how bad the problem is; it's related to some serious test flakes, though, so it's undetermined 
whether it's a substantial issue or a problem with the test.

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubectl completion failed to list file names](https://github.com/kubernetes/kubernetes/issues/60517)

## Green

Issues with an approved PR which is just waiting for labels, release notes, or automation.

* [Support out-of-tree authentication providers](https://github.com/kubernetes/kubernetes/issues/55968)
* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)
* [DaemonSet should ignore the `unschedulable` field of a node](https://github.com/kubernetes/kubernetes/issues/60163)
* [https://github.com/kubernetes/kubernetes/issues/60381](https://github.com/kubernetes/kubernetes/issues/60381)
* [[failing test] daemonset-upgrade in 1.9-master upgrade jobs](https://github.com/kubernetes/kubernetes/issues/60437)
* [Failing to list pods with selector `spec.nodeName!= ""`](https://github.com/kubernetes/kubernetes/issues/60507)
* [[failing test] Nginx should conform to Ingress spec](https://github.com/kubernetes/kubernetes/issues/60588)
* [[job flake] kubelet-master](https://github.com/kubernetes/kubernetes/issues/60604)
* [Mount propagation moved to beta, comment not updated](https://github.com/kubernetes/kubernetes/issues/60657)

## Kick

Issues just waiting for the grace period to elapse before being kicked out of 1.10.

* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)
* [Improve k8s support for multizone PDs](https://github.com/kubernetes/kubernetes/issues/59988) 

## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)

## Kick

Issues which are waiting for automation to kick them out of the milestone.

* [e2e tests for cloud-controller-manager](https://github.com/kubernetes/kubernetes/issues/44975)
* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)

**Comment by jberkus** [comment]
As of around 10am PST today, we have 25 issues open against the 1.10 milestone, which is an increase of 4 from Friday.  Most of the new issues are actually breakouts of a larger test fail issue, in order to have one issue per SIG for resolution (see test fails below).

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) 
and is up to date as of this morning. 

## Red

Issues with no PR, or no complete PR, which cannot be easily taken out of 1.10 or represent major regressions.

* [Kubernetes is vulnerable to stale reads, violating critical pod safety guarantees](https://github.com/kubernetes/kubernetes/issues/59848)

The Stale Reads issue is a potentially major issue, affecting all supported versions of Kubernetes, without a clear solution that
doesn't produce a major performance regression.  Depending on how fixing it goes, we may have to punt on it for 1.10.0 and wait for
a fix in 1.10.1.

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)

This shows what may be a very large performance regression in 1.10 even without a Stale Reads fix.  Test flakes are fixed, so hopefully we can get confirmation (or not).

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubeadm v1.10 release tracking issue](https://github.com/kubernetes/kubernetes/issues/60608)

### New Test Fails

We've added a bunch of test fail tracking over the weekend.  These represent related test fails across several test suites, with individual asignees.  All are considered Yellow right now, as they either have fixes in progress, or are too new to be considered stuck.  Most of these are being tracked from issue #60003.

* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [HPA is refuses to autoscale if DesiredReplicas > MaxReplicas (but I want it to scale up to MaxReplicas)](https://github.com/kubernetes/kubernetes/issues/59888)
* [[test failed] Services should work after restarting apiserver](https://github.com/kubernetes/kubernetes/issues/60761)
* [[test failed] regular resource usage tracking resource tracking for 0 pods per node](https://github.com/kubernetes/kubernetes/issues/60762)
* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[test failed] [1.10 upgrade] Servers with support for Table transformation](https://github.com/kubernetes/kubernetes/issues/60765)
* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade] Kubernetes Dashboard should check that the kubernetes-dashboard instance is alive](https://github.com/kubernetes/kubernetes/issues/60767)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)
* [Flexvolume e2e tests failing](https://github.com/kubernetes/kubernetes/issues/60725)
* [Advanced Audit tests flaking](https://github.com/kubernetes/kubernetes/issues/60719)

## Green

Issues with an approved PR which is just waiting for labels, release notes, or automation.

* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)
* [[failing test] daemonset-upgrade in 1.9-master upgrade jobs](https://github.com/kubernetes/kubernetes/issues/60437)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [Mount propagation moved to beta, comment not updated](https://github.com/kubernetes/kubernetes/issues/60657)
* [Failure in Conformance test - Daemon set [Serial] should update pod when spec was updated and update strategy is RollingUpdate [Conformance]](https://github.com/kubernetes/kubernetes/issues/60745)

## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)

## Kick

Issues just waiting for the grace period to elapse before being kicked out of 1.10.

* [Request to add namespace name and namespace UUID to metadata of on-disk log file](https://github.com/kubernetes/kubernetes/issues/58638)
* [Improve k8s support for multizone PDs](https://github.com/kubernetes/kubernetes/issues/59988)

**Comment by jberkus** [comment]
As of around 11am PST today, we have 26 issues open against the 1.10 milestone, which is an increase of 1 from yesterday.  Most issues are various test failures, as non-test-fail issues are getting resolved.

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) 
and is up to date as of this morning. 

## Red

Issues with no PR, or no complete PR, which cannot be easily taken out of 1.10 or represent major regressions.

* [Kubernetes is vulnerable to stale reads, violating critical pod safety guarantees](https://github.com/kubernetes/kubernetes/issues/59848)

The Stale Reads issue is a potentially major issue, affecting all supported versions of Kubernetes, without a clear solution that
doesn't produce a major performance regression.  It does not look to be headed towards resolution on a reasonable timeline for 1.10, so we may need to talk about it for 1.10.1.

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)

This shows what may be a very large performance regression in 1.10 even without a Stale Reads fix.  SIG thinks we have a major regression here, even though test flakes are making it hard to verify.

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

* [Improve k8s support for multizone PDs](https://github.com/kubernetes/kubernetes/issues/59988)  (this is an exception)

### Test Fails In Progress

The vast majority of issues open now are test fails.  All of the below are in progress in some way, but don't yet have a clear resolution.  Many of these are the usual upgrade test failures.  In at least one case, we need to unfreeze the 1.9 tree to fix the test.  Not clear at this point whether we have a general upgrade issue the way we did in 1.9.

* [[job failure] ci-kubernetes-e2e-kubeadm-gce](https://github.com/kubernetes/kubernetes/issues/59762)
* [[job failure] periodic-kubernetes-e2e-kubeadm-gce-selfhosting](https://github.com/kubernetes/kubernetes/issues/59766)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [Advanced Audit tests flaking](https://github.com/kubernetes/kubernetes/issues/60719)
* [Flexvolume e2e tests failing](https://github.com/kubernetes/kubernetes/issues/60725)
* [[test failed] Services should work after restarting apiserver](https://github.com/kubernetes/kubernetes/issues/60761)
* [[test failed] regular resource usage tracking resource tracking for 0 pods per node](https://github.com/kubernetes/kubernetes/issues/60762)
* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)
* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade] Kubernetes Dashboard should check that the kubernetes-dashboard instance is alive](https://github.com/kubernetes/kubernetes/issues/60767)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)

## Green

Issues with an approved PR which is just waiting for labels, release notes, or automation.

* [Kubelet often fails on AWS spot instances](https://github.com/kubernetes/kubernetes/issues/57382)
* [[failing test] daemonset-upgrade in 1.9-master upgrade jobs](https://github.com/kubernetes/kubernetes/issues/60437)
* [Mount propagation moved to beta, comment not updated](https://github.com/kubernetes/kubernetes/issues/60657)
* [Failure in Conformance test - Daemon set [Serial] should update pod when spec was updated and update strategy is RollingUpdate [Conformance]](https://github.com/kubernetes/kubernetes/issues/60745)
* [[test failed] [1.10 upgrade] Servers with support for Table transformation](https://github.com/kubernetes/kubernetes/issues/60765)

## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083) (just waiting on fixing the tests)
* [Bringing server-side printing to beta summary](https://github.com/kubernetes/kubernetes/issues/58536) (should be done?)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)

**Comment by jberkus** [comment]
As of around 4pm PST today, we have 18 issues open against the 1.10 milestone, which is a decrease of 4 from Wednesday, and a great trajectory to be on.  Most issues are various test failures, as non-test-fail issues are getting resolved, and all test fails are now getting attention.

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) 
and is up to date as of this afternoon.

Critical concerns right now are:

1. Potential major performance regression
2. Opening the 1.9 tree in order to patch the upgrade tests (see Yellow)

## Red

Issues with no PR, or no complete PR, which cannot be easily taken out of 1.10 or represent major regressions.

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)

This shows what may be a very large performance regression in 1.10 even without a Stale Reads fix.  SIG thinks we have a major regression here, even though test flakes are making it hard to verify.

* [Kubernetes is vulnerable to stale reads, violating critical pod safety guarantees](https://github.com/kubernetes/kubernetes/issues/59848)

The Stale Reads issue is still potentially major issue, affecting all supported versions of Kubernetes, without a clear solution that doesn't produce a major scalability regression.  However, it looks highly unlikely to be resolved in the next 2 weeks, so we are recommending taking it out of the 1.10 milestone.

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

### Test Fails In Progress

The vast majority of issues open now are test fails.  All of the below are in progress in some way, but don't yet have a clear resolution.  Many of these are the usual upgrade test failures. 

* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[test failed] regular resource usage tracking resource tracking for 0 pods per node](https://github.com/kubernetes/kubernetes/issues/60762)
* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)
* [k8s.io/kubernetes/examples test failed](https://github.com/kubernetes/kubernetes/issues/60905)

These four test fails seem to require modifying tests/code for 1.9 in order to fix.  We need a hotfix to allow the owners to do this, and then we need a better procedure for handling upgrade tests in the future so that it doesn't lead to needing to patch tests on an older, frozen, version.  Either that, or we have to decide to ignore the upgrade tests for 1.10.

* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade] Kubernetes Dashboard should check that the kubernetes-dashboard instance is alive](https://github.com/kubernetes/kubernetes/issues/60767)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)

## Green

Issues with an approved PR which is just waiting for labels, release notes, or automation.

* [Flexvolume e2e tests failing](https://github.com/kubernetes/kubernetes/issues/60725)
* [[test failed] Services should work after restarting apiserver](https://github.com/kubernetes/kubernetes/issues/60761)
* [FlexVolume probe race condition potentially crashes kubelet](https://github.com/kubernetes/kubernetes/issues/60861)
* ["Cluster level logging implemented by Stackdriver should ingest events" fails for GKE Regional Clusters](https://github.com/kubernetes/kubernetes/issues/60882)

## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083) (just waiting on fixing the tests)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)

**Comment by jberkus** [comment]
As of around 1pm PDT today, we have 15 accepted issues open against the 1.10 milestone, and four possibles, which is a decrease of 3 from Friday.  Further, four issues are likely to close in the next 2-5 hours as fixed tests pass.

However, we have a couple of major blocker issues which look possible to delay the release, see Red below.

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) 
and is up to date as of 1pm PDT. 

## Red

The big potential release-derailer is the major performance regressions possibly due to unidentified performance changes in etcd:

* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)

At this point, it is unclear if the etcd issues account for all of the problems, or if changing etcd version/settings will fix the issues.

This is an apparently unrelated increase on memory used by the API server.  @shyamjvs has been hard at work bisecting for this, and may have found a culprit:

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)

IMHO, these two performance regressions are significant enough to warrant a release delay.

We also have two test fails which have been receiving no attention.  While neither looks that bad, I'm flagging them because we don't actually know what's causing them:

* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)

The Stale Reads issue has been removed from 1.10.

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

* None.  Yay!  But ...

### Test Fails In Progress

All of the below are in progress in some way, but don't yet have a clear resolution.  

* [[test failed] regular resource usage tracking resource tracking for 0 pods per node](https://github.com/kubernetes/kubernetes/issues/60762)
* [pull-kubernetes-kubemark-e2e-gce is failing](https://github.com/kubernetes/kubernetes/issues/60870)

## Possibles

These three issues may be 1.10 issues; they were recently reported and seem related to other issues with 1.10. However, none of them have been examined by the SIGs yet.  None look like release-blockers.

* [k8s.io/kubernetes/examples test failed](https://github.com/kubernetes/kubernetes/issues/60905)
* [zsh completion throws error v1.10.0-beta.2](https://github.com/kubernetes/kubernetes/issues/61017)
* [KUBE_GIT_VERSION contains extraneous comma when building from git archive source.tgz](https://github.com/kubernetes/kubernetes/issues/61029)
* [Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)


## Green

Issues with an approved PR which is just waiting for labels, release notes, or automation.  A bunch of these are tests which are being fixed now that code was merged into 1.9, just waiting on them to pass.

* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)
* [FlexVolume probe race condition potentially crashes kubelet](https://github.com/kubernetes/kubernetes/issues/60861)
* ["Cluster level logging implemented by Stackdriver should ingest events" fails for GKE Regional Clusters](https://github.com/kubernetes/kubernetes/issues/60882)

## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083) (waiting on docks merge)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)

**Comment by jberkus** [comment]
Burndown report as of 10am PDT March 13


## Red

Issues which are blockers, or are status unknown and look serious, without a good PR.

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)
* [[test failed] gci-gce-alpha-features](https://github.com/kubernetes/kubernetes/issues/61050)

## Yellow

Issues which are blockers, with a good PR.  Also undecided issues with or without PRs which look like they won't be considered 1.10 bugs.

* [Schedule DaemonSet Pods by default scheduler](https://github.com/kubernetes/kubernetes/issues/59194)
* [pull-kubernetes-kubemark-e2e-gce is failing](https://github.com/kubernetes/kubernetes/issues/60870)
* ["Cluster level logging implemented by Stackdriver should ingest events" fails for GKE Regional Clusters](https://github.com/kubernetes/kubernetes/issues/60882)
* [zsh completion throws error v1.10.0-beta.2](https://github.com/kubernetes/kubernetes/issues/61017)
* [Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)
* ["CreateContainerConfigError: failed to prepare subPath for volumeMount" error with configMap volume](https://github.com/kubernetes/kubernetes/issues/61076)
* [Subpath tests don't work in multizone GCE](https://github.com/kubernetes/kubernetes/issues/61101)

## Green

Non-blocker issues.

* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)
* [HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)

## Tracking Issues

* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubeadm v1.10 release tracking issue](https://github.com/kubernetes/kubernetes/issues/60608)

**Comment by jberkus** [comment]
Corrected March 13th Burndown report:

Status as of noon, PDT, March 13th.  

We still have 19 issues open.  Some of these don't show up on github searches because they are not labelled correctly, which is being resolved.

We are waiting for a bunch of changes to go into 1.9 to fix downgrade tests, which should clear up some of the pending issues list.  This does point to needing a change in how we do downgrade tests in the future.

There are several major regressions without resolution right now, sufficient to delay ending code freeze, and possibly the final release per burndown meeting this AM.

## Red

Issues which are blockers, or are status unknown and look serious, without a good PR.

Performance issues in analysis/progress:

* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)

Failing tests with currently unknown causes:

* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)
* [[test failed] gci-gce-alpha-features](https://github.com/kubernetes/kubernetes/issues/61050)

Regression in progress, but fix untested.  Issue was accidentally dropped by SIG and just picked up again:

* [Kubelet leaves orphaned directories and mounts after downtime](https://github.com/kubernetes/kubernetes/issues/60645)

## Yellow

Issues which are blockers, with a good PR.  Also undecided issues with or without PRs which look like they won't be considered 1.10 bugs.

* [Schedule DaemonSet Pods by default scheduler](https://github.com/kubernetes/kubernetes/issues/59194)


### Test Fails in Progress

* ["Cluster level logging implemented by Stackdriver should ingest events" fails for GKE Regional Clusters](https://github.com/kubernetes/kubernetes/issues/60882)
* ["CreateContainerConfigError: failed to prepare subPath for volumeMount" error with configMap volume](https://github.com/kubernetes/kubernetes/issues/61076)
* [Subpath tests don't work in multizone GCE](https://github.com/kubernetes/kubernetes/issues/61101)

### 1.9 downgrade tests issues

These issues are waiting for code on 1.9 in order to fix the downgrade tests.

* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)
* [pull-kubernetes-kubemark-e2e-gce is failing](https://github.com/kubernetes/kubernetes/issues/60870)

## Green

Non-blocker issues.

* [zsh completion throws error v1.10.0-beta.2](https://github.com/kubernetes/kubernetes/issues/61017)
* [Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)
* [HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)

## Tracking Issues

* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubeadm v1.10 release tracking issue](https://github.com/kubernetes/kubernetes/issues/60608)

**Comment by jberkus** [comment]
Status as of noon, PDT, March 14th.  

We have three issues which were re-opened yesterday, because they were closed in advance of verifying the tests.

Several of the upgrade/downgrade tests have been fixed, but we are waiting on all tests to pass before we actually clear them, since several test fails have been closed and reopened. A big thanks to @liggit for pursuing those. 

Overall status is "Crimson".  We have multiple unclosed issues, any of which are sufficient to block release, and two of which (performance/scalability and ) have no specific timeline for resolution.   We also have one trailing feature of unknown status.  Further delaying the release seems more likely than not. 

## Red

Issues which are blockers, or are status unknown and look serious, without a good PR.

Performance issues in analysis/progress. Some of the performance and scalability issues have been resolved, and others have been broken out into more specific issues.  There are some issues (see Green below) which are not expected to be resolved for 1.10, but are regarded as non-blockers.  Many thanks to @shyamjvs for diving into these regressions!

* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [Fluentd-scaler causing fluentd pod deletions and messes with ds-controller](https://github.com/kubernetes/kubernetes/issues/61190)

Failing tests with currently unknown causes:

* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[test failed] gci-gce-alpha-features](https://github.com/kubernetes/kubernetes/issues/61050)

Regression in progress, but fix not passing tests:

* [Kubelet leaves orphaned directories and mounts after downtime](https://github.com/kubernetes/kubernetes/issues/60645)

Orphaned feature, awaiting response from SIG:

* [Schedule DaemonSet Pods by default scheduler](https://github.com/kubernetes/kubernetes/issues/59194)

## Yellow

Issues which are blockers, with a good PR.  Also undecided issues with or without PRs which look like they won't be considered 1.10 bugs.

### Test Fails in Progress

These are currently all upgrade tests

* [[test failed] [1.10 upgrade] Servers with support for Table transformation](https://github.com/kubernetes/kubernetes/issues/60765)
* [[test failed] [1.10 upgrade][gci-gke] Kubernetes Dashboard should check that the kubernetes-dashboard instance is alive](https://github.com/kubernetes/kubernetes/issues/60767)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)


## Green

Non-blocker issues (expected to remain broken for 1.10, need to add release note):

[Flaky timeouts while waiting for RC pods to be running in density test](https://github.com/kubernetes/kubernetes/issues/61189)
[Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)
[HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)

Resolved, pending having all tests passing:

[Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
[[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
[pull-kubernetes-kubemark-e2e-gce is failing](https://github.com/kubernetes/kubernetes/issues/60870)
["Cluster level logging implemented by Stackdriver should ingest events" fails for GKE Regional Clusters](https://github.com/kubernetes/kubernetes/issues/60882)
["CreateContainerConfigError: failed to prepare subPath for volumeMount" error with configMap volume](https://github.com/kubernetes/kubernetes/issues/61076)
[Subpath tests don't work in multizone GCE](https://github.com/kubernetes/kubernetes/issues/61101)

Resolved, waiting for automation:

[zsh completion throws error v1.10.0-beta.2](https://github.com/kubernetes/kubernetes/issues/61017)

## Tracking Issues

* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubeadm v1.10 release tracking issue](https://github.com/kubernetes/kubernetes/issues/60608)

**Comment by jberkus** [comment]
As of around noon PDT today, we have 12 accepted issues open against the 1.10 milestone, which is a decrease of 3 from yesterday.  Further, four issues are likely to close in the next 2-5 hours as fixed tests pass.

However, we have a couple of major blocker issues which look possible to delay the release, see Red below.

Tracking spreadsheet is [here](https://docs.google.com/spreadsheets/d/1qhgPfH99Vz-rRp_yMV2YdGlijDoa24d59cuo89Ebpyg/edit#gid=2058989556) 
and is up to date as of 1pm PDT. 

## Red

The big potential release-derailer is the major performance regressions:

* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [Fluentd-scaler causing fluentd pod deletions and messes with ds-controller](https://github.com/kubernetes/kubernetes/issues/61190)
* [Flaky timeouts while waiting for RC pods to be running in density test](https://github.com/kubernetes/kubernetes/issues/61189) (non-blocker)

These two performance regressions are considered significant enough to warrant a release delay work on them, including git bisect and scalability testing, has been ongoing.  This is slow due to the relatively small number of folks who understand kube scalability and the scalability tests.  You a performance geek who wants to get involved with Kubernetes?  We could use you.

This test fail may be related to the fluentd performance issues, but root cause unknown:

* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)

Pod deletion has a problem with race conditions, work is in progress but intial patch attempt needs work:

* [Kubelet leaves orphaned directories and mounts after downtime](https://github.com/kubernetes/kubernetes/issues/60645)

## Yellow

Issues with a PR that is not approved, or issues with no PR which look possible to ignore/kick in 1.10.

The problem with being unable to delete PVCs on downgrade is in progress, in the form of manual downgrade docs and a patch for the tests, with an actual fix due in 1.9.5:

* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)

This GCE deprecated flag issue has a PR in progress and near approval.  It is currently breaking a lot of unrelated tests:

* [cluster/gce uses deprecated gcloud parameter "mode"](https://github.com/kubernetes/kubernetes/issues/54238)

The rest of the Daemonset Scheduling work *looks* likely to be postponed until 1.11, but it's unclear at this point what would be required to back out committed work:

* [Schedule DaemonSet Pods by default scheduler](https://github.com/kubernetes/kubernetes/issues/59194)

## Green

### Issues that are non-blockers or expected regressions and are expected to remain issues after 1.10.0 release:

* [HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)
* [Apiserver CPU/Mem usage bumped to 1.5-2x in big clusters](https://github.com/kubernetes/kubernetes/issues/60500)
* [Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)

### Issues with an approved PR which is just waiting for labels, release notes, or automation:

* ["CreateContainerConfigError: failed to prepare subPath for volumeMount" error with configMap volume](https://github.com/kubernetes/kubernetes/issues/61076)

### Test fails which have been fixed but we're waiting for a couple days of green before we stop watching them:

* [Advanced Audit tests flaking](https://github.com/kubernetes/kubernetes/issues/60719)
* [[test failed] [1.10 upgrade] Servers with support for Table transformation](https://github.com/kubernetes/kubernetes/issues/60765)
* [[test failed] [1.10 upgrade]  Dynamic Provisioning DynamicProvisioner](https://github.com/kubernetes/kubernetes/issues/60766)
* [[test failed] [1.10 upgrade][gci-gke] Kubernetes Dashboard should check that the kubernetes-dashboard instance is alive](https://github.com/kubernetes/kubernetes/issues/60767)
* [[test failed] [1.10 upgrade] Cadvisor should be healthy on every node](https://github.com/kubernetes/kubernetes/issues/60768)
* [[test failed] [1.10 upgrade] Proxy version v1](https://github.com/kubernetes/kubernetes/issues/60769)
* [pull-kubernetes-kubemark-e2e-gce is failing](https://github.com/kubernetes/kubernetes/issues/60870)
* ["Cluster level logging implemented by Stackdriver should ingest events" fails for GKE Regional Clusters](https://github.com/kubernetes/kubernetes/issues/60882)
* [Subpath tests don't work in multizone GCE](https://github.com/kubernetes/kubernetes/issues/61101)


## Special Issues

Primarily tracking issues.

* [v1.10 known issues / FAQ accumulator ](https://github.com/kubernetes/kubernetes/issues/59764)
* [Advanced Auditing 1.10 umbrella bug](https://github.com/kubernetes/kubernetes/issues/58083) (waiting on docks merge)
* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)

**Comment by jberkus** [comment]
Status as of noon, PDT, March 19th. 

Overall status is "saffron" (yellow with some organge).  While the majority of bugs are either closed or have short-timeline plans for closure, we still have outstanding performance issue(s) whose cause is unknown.

## Red

Issues which are blockers, or are status unknown and look serious, without a good PR.

Performance issues in analysis/progress. With fluentd patches, performance issues have been addressed within acceptable tolerances (there is increased resource usage in this version of Kubernetes, period).  Except this one, whose cause is still unknown:

* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [Fluentd-scaler causing fluentd pod deletions and messes with ds-controller](https://github.com/kubernetes/kubernetes/issues/61190) (this was patched, but the patch did not improve test results)

## Yellow

Issues which are blockers, with a good PR.  Also undecided issues with or without PRs which look like they won't be considered 1.10 bugs.

### Features

* [Schedule DaemonSet Pods by default scheduler](https://github.com/kubernetes/kubernetes/issues/59194) (merged except for Docs/Release Notes)

PVC protection:

* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)

This is being dealt with as a documentation bug with a documented workaround.  There is a patch in progress against 1.9 that will make PVC downgrade work, to come out with 1.9.6.  In the meantime, users who have a lot of PVCs should be encouraged to wait to upgrade until 1.9.6 is out.

### Bugs

* [Current image in glbc.manifest points to alpha version](https://github.com/kubernetes/kubernetes/issues/61305) (gave go ahead at Release Burndown)

### Test Fails in Progress

* [[failing test] should restart all nodes and ensure all nodes and pods recover](https://github.com/kubernetes/kubernetes/issues/60763)
* [[test failed] gci-gce-alpha-features](https://github.com/kubernetes/kubernetes/issues/61050)

Both of these test fails are related to either Daemonset scheduling, and should turn green soon now that PRs are merged.  Hoping!

## Green

Non-blocker issues (expected to remain broken for 1.10, generally need to add release note):

[Flaky timeouts while waiting for RC pods to be running in density test](https://github.com/kubernetes/kubernetes/issues/61189)
[Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)
[HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)
[kubeadm: etcd certs missing in self-hosted deploy](https://github.com/kubernetes/kubernetes/issues/61322) (will be fixed in point release)

## Tracking Issues

* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubeadm v1.10 release tracking issue](https://github.com/kubernetes/kubernetes/issues/60608)

**Comment by jberkus** [comment]
Status as of 11am, PDT, March 20th. 

Overall status is "tangerine" (trending red).  While the majority of bugs are either closed, we still have outstanding performance issue(s) whose cause is unknown and may delay the release.  We also have several other unrelated issues which need fixing.

## Red

Release Blockers without a resolution timeline of less than 24 hours.

Performance issues in analysis/progress. There are two, which may be related, causing unacceptable performance on GCE.  The cause of these may be in some way related to fluentd, but that doesn't make them a non-blocker:

* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)
* [Fluentd-scaler causing fluentd pod deletions and messes with ds-controller](https://github.com/kubernetes/kubernetes/issues/61190) (this was patched, but the patch did not improve test results)

GKE tests are no longer running due to some issue with GKE, but we need this resolved before we release:

* [All gke OSS tests are down since ~8p yesterday?](https://github.com/kubernetes/kubernetes/issues/61416)

## Yellow

Issues which are blockers, which are expected to resolve in 24 hours.  Also undecided issues with or without PRs which look like they won't be considered 1.10 bugs.

Daemonset Scheduling feature needs to be reverted, or more accurately neutralized by disabling the alpha gate, before release:

* [DaemonSet scheduling conflates hostname and nodename](https://github.com/kubernetes/kubernetes/issues/61410)
* [[test failed] gci-gce-alpha-features](https://github.com/kubernetes/kubernetes/issues/61050)
* [Schedule DaemonSet Pods by default scheduler](https://github.com/kubernetes/kubernetes/issues/59194)

PVC protection workaround for 1.10.0, with fix pending for 1.9.  This shouldn't be a blocker anymore:

* [[job failed] 1.9-master upgrade|downgrade jobs](https://github.com/kubernetes/kubernetes/issues/60764)

## Green

Non-blocker issues (expected to remain broken for 1.10, generally need to add release note):

[Flaky timeouts while waiting for RC pods to be running in density test](https://github.com/kubernetes/kubernetes/issues/61189)
[Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)
[HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)
[kubeadm: etcd certs missing in self-hosted deploy](https://github.com/kubernetes/kubernetes/issues/61322) (will be fixed in point release)
[Mounting socket files from subPaths fail](https://github.com/kubernetes/kubernetes/issues/61377)

## Tracking Issues

* [[job failure]ci-kubernetes-e2e-gci-gke|gce-serial](https://github.com/kubernetes/kubernetes/issues/60003)
* [kubeadm v1.10 release tracking issue](https://github.com/kubernetes/kubernetes/issues/60608)

**Comment by jberkus** [comment]
Status as of 11am, PDT, March 21st.  Happy Nowruz!  "zardi ye man az to, sorkhi ye to az man" seems particularly appropriate here. 

Overall status is "straw" (light yellow).  At this point, everthing is resolved or resolving in the next few hours except for #60589, which suffers from having to make a potentially painful tradeoff.

## Red

Release Blockers without a resolution timeline of less than 24 hours.

This issue has been traced to a commit which was also a bugfix.  At this point, we need opinions from multiple SIG leads about what to do on a reversion:

* [[test flakes] master-scalability suites](https://github.com/kubernetes/kubernetes/issues/60589)

## Yellow

Issues which are blockers, which are expected to resolve in 24 hours.  Also undecided issues with or without PRs which look like they won't be considered 1.10 bugs.

Issue with subpaths which would prevent someone from upgrading from specific versions of Kubernetes, which has a patch just waiting to be cherrypicked:

* [Issues with using OuterVolumeSpecName for Volume Subpaths](https://github.com/kubernetes/kubernetes/issues/61372)

## Green

The Fluentd scaler issue has been fixed sufficient to not be a blocker for 1.10.  There are still effects of it which will need fixing in future point releases:

* [Fluentd-scaler causing fluentd pod deletions and messes with ds-controller](https://github.com/kubernetes/kubernetes/issues/61190) (this was patched, but the patch did not improve test results)

GKE tests are now running.

Daemonset scheduling and PVC protection issues have been resolved.  Important release note for PVC protection regarding downgrades.


Non-blocker issues (expected to remain broken for 1.10, generally need to add release note):

[Flaky timeouts while waiting for RC pods to be running in density test](https://github.com/kubernetes/kubernetes/issues/61189)
[Controller-manager sees higher mem-usage when load test runs before density](https://github.com/kubernetes/kubernetes/issues/61041)
[HostPath mounts failing with "Path is not a shared or slave mount"](https://github.com/kubernetes/kubernetes/issues/61058)
[kubeadm: etcd certs missing in self-hosted deploy](https://github.com/kubernetes/kubernetes/issues/61322) (will be fixed in point release)
[Mounting socket files from subPaths fail](https://github.com/kubernetes/kubernetes/issues/61377)

**Comment by jberkus** [comment]
1.10 is out now, closing.
